pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java

```diff
@@ -15,66 +15,105 @@
     _controllerConf = controllerConf;
+
+    // Pick the correct scheme for PinotFS. For pluggable storage, we will expect that the directory is configured
+    // with a storage prefix. For backwards compatibility, we will assume local/mounted NFS storage if nothing is configured.
+    String scheme = "file";
+    String dataDir = _controllerConf.getDataDir();
+    try {
+      URI uri = new URI(_controllerConf.getDataDir());
+      scheme = uri.getScheme();
+    } catch (Exception e) {
+      // Assume local FS
+      LOGGER.warn("Could not create controller config data dir URI with {}", _controllerConf.getDataDir());
+      dataDir = "file://" + _controllerConf.getDataDir();
+    }
+
+    PinotFS pinotFS = PinotFSFactory.create(scheme);
+
     try {
-      _baseDataDir = new File(_controllerConf.getDataDir());
-      if (!_baseDataDir.exists()) {
-        FileUtils.forceMkdir(_baseDataDir);
+      _baseDataDirURI = new URI(dataDir);
+      if (!pinotFS.exists(_baseDataDirURI)) {
+        pinotFS.mkdir(_baseDataDirURI);
       }
-      _fileUploadTmpDir = new File(_baseDataDir, "fileUploadTemp");
-      if (!_fileUploadTmpDir.exists()) {
-        FileUtils.forceMkdir(_fileUploadTmpDir);
+      _fileUploadTmpDirURI = new URI(_baseDataDirURI + "/fileUploadTemp");
+      if (!pinotFS.exists(_fileUploadTmpDirURI)) {
+        pinotFS.mkdir(_fileUploadTmpDirURI);
       }
-      _tmpUntarredPath = new File(_fileUploadTmpDir, "untarred");
-      if (!_tmpUntarredPath.exists()) {
-        _tmpUntarredPath.mkdirs();
+      _tmpUntarredPathURI = new URI(_fileUploadTmpDirURI + "/untarred");
+      if (!pinotFS.exists(_tmpUntarredPathURI)) {
+        pinotFS.mkdir(_tmpUntarredPathURI);
       }
-      _schemasTmpDir = new File(_baseDataDir, "schemasTemp");
-      if (!_schemasTmpDir.exists()) {
-        FileUtils.forceMkdir(_schemasTmpDir);
+      _schemasTmpDirURI = new URI(_baseDataDirURI + "/schemasTemp");
+      if (!pinotFS.exists(_schemasTmpDirURI)) {
+        pinotFS.mkdir(_schemasTmpDirURI);
       }
       _vip = _controllerConf.generateVipUrl();
     } catch (Exception e) {
       throw new InvalidControllerConfigException("Bad controller configuration", e);
     }
   }
 
+  public String getVip() {
+    return _vip;
+  }
+
+  public URI getFileUploadTmpDirURI() {
+    return _fileUploadTmpDirURI;
+  }
+
+  public URI getBaseDataDirURI() {
+    return _baseDataDirURI;
+  }
+
+  public URI getTmpUntarredPathURI() {
+    return _tmpUntarredPathURI;
+  }
+
+  public URI getSchemasTmpDirURI() {
+    return _schemasTmpDirURI;
+  }
+
+  // TODO: The following getters that return files should eventually be migrated to URIs once pluggable storage support
+  // is complete throughout the system. Leaving this here for backwards compatibility for now.
   public File getFileUploadTmpDir() {
-    return _fileUploadTmpDir;
+    return new File(_fileUploadTmpDirURI);
```

Comment:
This seems like a change that could impact existing code (ie we are creating a new file from URI). Are there tests covering this api? If not, please add to ensure that the change does not break existing functionality.